pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -1961,6 +2010,55 @@
           %leave  ~
       ==
     --
+  ::  +add-cache: remember a cache mapping
+  ::
+  ++  add-cache
+    |=  [=path =cache-entry]
+    ^-  [(list move) server-state]
+    =/  cache-entries  (fall (~(get of cache.state) path) ~)
+    =.  cache.state  (~(put of cache.state) path [[~ cache-entry] cache-entries])
+    `state
+  ::  +del-cache: forget a cache mapping
+  ::
+  ++  del-cache
+    |=  [=path ucase=(unit @ud)]
+    ^-  [(list move) server-state]
+    =/  cache-entries  (fall (~(get of cache.state) path) ~)
+    ?~  cache-entries
+      `state
+    ::  no case means the most recent entry
+    ::
+    =/  case=@ud  (fall ucase (sub (lent cache-entries) 1))
+    ?:  (gte case (lent cache-entries))
+      `state
+    ::  replace the entry with ~
+    ::  and give a %cache-changed for each affected binding
+    ::
+    =.  cache-entries  (snap cache-entries case `(unit cache-entry)`~)
```

Comment:
feedback appreciated on this code -- it looks kinda ugly to me, so I suspect there's a more idiomatic approach